data_rentgen/server/services/lineage.py

```diff
@@ -197,7 +202,22 @@ async def get_lineage_by_jobs(  # noqa: C901
                 (dataset_symlink.from_dataset_id, dataset_symlink.to_dataset_id): dataset_symlink
                 for dataset_symlink in dataset_symlinks
             }
-
+        column_lineage_dataset_relations = await self._uow.column_lineage.list_by_job_ids(
+            sorted(result.jobs.keys()),
+            since,
+            until,
+        )
+        fingerprints = []
+        result.column_lineage = defaultdict(list)
+        for dataset_relation in column_lineage_dataset_relations:
+            result.column_lineage[(dataset_relation.target_dataset_id, dataset_relation.source_dataset_id)].append(
+                dataset_relation,
+            )
+            fingerprints.append(dataset_relation.fingerprint)
+        column_relations = await self._uow.dataset_column_relation.list_column_relations_by_fingerprint(fingerprints)
+        result.column_relations = defaultdict(list)
+        for column_relation in column_relations:
+            result.column_relations[column_relation.fingerprint].append(column_relation)
```

Comment:
Why not using `JOIN`? For example:
```sql
SELECT max(cl.created_at), cl.source_dataset_id, cl.target_dataset_id, dcr.source_column, dcr.target_column, bit_or(dcr.type)
FROM column_lineage AS cl
JOIN dataset_column_relations AS dcr ON dcr.fingerprint = cl.fingerprint
WHERE
    cl.created_at >= ...
AND cl.operation_id = ANY(...)
```

This already returns aggregated information for the response, and only thing that should be done in Python is converting types from bit field sum to list of strings.